daemon/api.go

```diff
@@ -634,21 +638,29 @@ func (inst *snapInstruction) Agreed(intro, license string) bool {
 var snappyInstall = snappy.Install
 
 func (inst *snapInstruction) install() interface{} {
+	// FIXME: handle license agreement
 	flags := snappy.DoInstallGC
 	if inst.LeaveOld {
 		flags = 0
 	}
-	_, err := snappyInstall(inst.pkg, inst.Channel, flags, inst)
-	if err != nil {
-		if inst.License != nil && snappy.IsLicenseNotAccepted(err) {
-			return inst.License
-		}
-		return err
-	}
+	state := inst.overlord.StateEngine().State()
+	state.Lock()
+	chg := state.NewChange("install-snap", fmt.Sprintf("Installing %s from %s", inst.pkg, inst.Channel))
+	err := snapstate.Install(chg, inst.pkg, inst.Channel, flags)
+	state.Unlock()
 
-	// TODO: return the log
-	// also to do: commands update their output dynamically, as it changes.
-	return nil
+	inst.overlord.SnapManager().Ensure()
+	inst.overlord.SnapManager().Wait()
```

Comment:
these can only be very temporary. We would need Wait etc at the overlord level also we don't have a way to ask synchronously for an Ensure on the overlord though, the most we have is EnsureBefore which is async